### ISSUE TITLE
different appearance of one day to mutiple day events

### ISSUE THREAD

**Issue by fmuell2** [marcusolsson/grafana-calendar-panel#24]
Currently one day events just get a colored bubble in front of the eventtitle. Mutiple day events are colored the entire eventitle. In my opinion, it would look nicer if one day events are colored the same way.

Maybe there should be panel setting to toggle between both modes for one day events.

**Comment by marcusolsson** [comment]
This is how both Google Calendar and Apple Calendar differentiates between single and multi-day events, both which I'm using for inspiration for the plugin. I'm open to discussing this, but I'd need a stronger argument for using the same styled would be better.

**Comment by fmuell2** [comment]
This came into my mind when playing with the multi-calender-query feature. We have many planned changes over multiple days highlighted in color a and some critical changes (mostly one day changes) in color b. In my opinion the less critical changes are highlighted more than the critical ones. They just seem to get lost in the crowd.

I think i have to discuss this with our internal costumers about their opinion. Maybe they come up with a crucial reason ;)

**Comment by marcusolsson** [comment]
Hm, I understand the issue and why a shared style would make sense here. 

I guess the question then is rather how do you'd differentiate between events of different priority. First thing that comes to mind is to use more muted colors for lower priority events, and more saturated colors for higher priority.

I have another feature request for manually setting the color for each calendar (https://github.com/marcusolsson/grafana-calendar-panel/issues/23). Do you thing that would help?

**Comment by fmuell2** [comment]
Yes, thats my point of view too. But i think even when you use a more saturated color for higher priority, if it only colors the little dot in front of the title, the event still looks not as important as a muti day event. I will try to make a sample calender dashboard and post it here.

**Comment by fmuell2** [comment]
![image](https://user-images.githubusercontent.com/84320356/120962967-13085380-c761-11eb-854b-cf07066a5e19.png)

For me, the multitimeevents look more important than the emergency change on 09.06.2021. The multiday color highlights the unimportant and normal events (but not because they are critical, only because they are multiday) more than the little dot in front of the emergency change. I am not sure if using a red instead of a light blue dot would help to highlight the emergency change enough.

**Comment by marcusolsson** [comment]
Yeah, it doesn't help that the text is a little too greyed out for single-day events unless you hover it.

I don't have any way of proving it, but I _think_ the majority of events will be single-day events, as it fits well with with event logs for example. My main worry is that by making single-day events more colorful, the calendar would overpower other panels.

Rather than making single-day events grab _more_ attention, could we change it so that multi-day events grab _less_ attention?

For example:

- Make colors for multi-day events more muted and boost the contrast of single-day events
- Experiment with more subtle styles for multi-day, for example remove fill and only have an outline

**Comment by fmuell2** [comment]
We really like the way multiday events look like now, it is exactly what we are looking for. But your suggestion with an outline instead of a whole title fill sounds promising too. I think there will be use cases for both stylings, so maybe adding a panel option for toggling between outline/complete fill will be good.

But my major point here is, that the appearance should not really differ between single and multiday as it does today. If you use complete fill both should be filled, if you use small outlines both types should use it.

**Comment by marcusolsson** [comment]
Thanks for all the great feedback! 🙏 

I think the next step would be to create some mock-ups for some possible solutions. We can always make this configurable as you say, e.g. "Display single day events as all-day events".

I'll update this issue with some screenshots after I've experimented a bit!

**Comment by dueni0** [comment]
A view like https://codepen.io/jaschulke/pen/qhGEp would be great. 
Even better with color chooser mentioned in #23
